client/src/services/apiService.ts

```diff
@@ -0,0 +1,77 @@
+import { Game } from '../@types/game'
+
+const BASE_URL = 'http://localhost:3000'
+
+export const getVenues = () => {
+  return fetchData('/venues')
+}
+
+export const getGames = () => {
+  return fetchData('/games')
+}
+
+export const getGame = (id: string) => {
+  return fetchData('/games/' + id)
+}
+
+interface JoinGame {
+  userName: string;
+}
+
+export const joinGame = (id: string, body: JoinGame) => {
+  return fetchData('/games/' + id, 'PUT', body)
+}
+
+export const createGame = (body: Game) => {
+  return fetchData('/games', 'POST', body)
+}
+
+export const getUser = (id: string) => {
+  return fetchData('/users/' + id)
+}
+
+export const getUsers = () => {
+  return fetchData('/users')
+}
+
+interface UserLogin {
+  usernameOrEmail: string;
+  password: string;
+}
+
+export const loginUser = (body: UserLogin) => {
+  return fetchData('/login', 'POST', body)
+}
+
+interface UserRegistration {
+  username: string;
```

Comment:
In RegisterForm.tsx this property is called userName instead. I think it needs to be changed either here or there...